server/api/discrepancies.py

```diff
@@ -45,11 +45,12 @@ def ballot_vote_deltas(
 
 
 def batch_vote_deltas(
-    reported_results: Dict[str, int], audited_results: Dict[str, int],
+    reported_results: macro.ChoiceVotes, audited_results: macro.ChoiceVotes
 ) -> Optional[ContestVoteDeltas]:
     deltas = {
         choice_id: reported_results[choice_id] - audited_results[choice_id]
         for choice_id in reported_results.keys()
+        if choice_id != "ballots"
```

Comment:
For my understanding, what is this check for?

Reply:
The macro module defined a data format for batch tallies that was a single dictionary where the keys were all the contest choice ids plus one special `"ballots"` key for the total number of ballots. Ideally, it would have been structurally separated (e.g. `{ "choices": {...votes by choice id...}, "ballots": number }`). Since it's flattened, this just works around that special key. It wasn't an issue in the previous PR because the code that consumed the vote deltas was iterating over a separate list of contest choices defined in the contest (implicitly ignoring `"ballots"`), whereas now there's some code that iterates over the keys of vote deltas directly.

Reply:
Got it, thanks for clarifying